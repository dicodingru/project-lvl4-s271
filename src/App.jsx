import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import { addMessage } from './actions';
import Root from './components/Root';

export default (store, socket) => {
  socket.on('newMessage', ({ data }) => {
    const { attributes } = data;
    store.dispatch(addMessage({ ...attributes }));
  });

  ReactDOM.render(
    <Provider store={store}>
      <Root />
    </Provider>,
    document.getElementById('chat')
  );
};
