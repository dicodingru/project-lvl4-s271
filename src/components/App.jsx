import React from 'react';
import PropTypes from 'prop-types';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { addMessage } from '../actions';
import rootReducer from '../reducers';
import Root from './Root';

/* eslint-disable no-underscore-dangle */
const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__;
/* eslint-enable */

const App = ({ initialState, socket }) => {
  const store = createStore(
    rootReducer,
    initialState,
    reduxDevtools && reduxDevtools()
  );

  socket.on('newMessage', ({ data }) => {
    const { attributes } = data;
    store.dispatch(addMessage({ ...attributes }));
  });

  return (
    <Provider store={store}>
      <Root />
    </Provider>
  );
};

App.propTypes = {
  initialState: PropTypes.shape({
    channels: PropTypes.array,
    messages: PropTypes.array
  }).isRequired,
  socket: PropTypes.instanceOf(WebSocket).isRequired
};

export default App;
