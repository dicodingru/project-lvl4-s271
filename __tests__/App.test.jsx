import React from 'react';
import App from '../src/components/App';

test('App', () => {
  const wrapper = mount(
    <App
      initialState={{
        channels: [
          { id: 1, name: 'general', removable: false },
          { id: 2, name: 'random', removable: false }
        ],
        messages: [
          {id: 1, channelId: 1, username: 'user1', text: 'message1'},
          {id: 2, channelId: 1, username: 'user2', text: 'message2'}
        ]
      }}
    />
  );

  expect(wrapper.render()).toMatchSnapshot();
});
