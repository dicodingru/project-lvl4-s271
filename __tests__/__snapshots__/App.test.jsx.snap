// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<div
  class="sc-gzVnrw kSwFRd"
>
  <div
    class="sc-bwzfXH jxfKnz"
  >
    <div
      class="sc-bdVaJa jWKkjw"
    >
      <a
        href="#general"
      >
        <span>
          general
        </span>
      </a>
    </div>
    <div
      class="sc-bdVaJa jWKkjw"
    >
      <a
        href="#random"
      >
        <span>
          random
        </span>
      </a>
    </div>
  </div>
  <div
    class="sc-bZQynM jWOpbX"
  >
    <div>
      <span>
        message1
      </span>
    </div>
    <div>
      <span>
        message2
      </span>
    </div>
    <div
      class="sc-htpNat daVAfG"
    >
      <form
        class="sc-bxivhb bUZGRr"
      >
        <input
          class="sc-ifAKCX jkqzbQ"
          type="text"
        />
        <button
          class="sc-EHOje dObCnd"
          type="submit"
        >
          Send
        </button>
      </form>
    </div>
  </div>
</div>
`;
